### ISSUE TITLE
Online project links problem in multilingual websites using the single-tree approach.

### ISSUE THREAD

**Issue by tutotio** [alkacon/opencms-core#497]
When you get a localized link into a page using CmsLocalePrefixLinkSubstitutionHandler (getLink(...) method in CmsDefaultLinkSubstitutionHandler) the link is stored in cache:

`exportManager.cacheOnlineLink(cacheKey, resultLink);` --> Line 298 in CmsDefaultLinkSubstitutionHandler

but cacheKey doesn't contain locale reference.

When you call again the same link for another locale, it finds it in cache:

`resultLink = exportManager.getCachedOnlineLink(cacheKey); `--> Line 272 in CmsDefaultLinkSubstitutionHandler

and you get the link but for the first locale instead of the current locale.

To solve it, you have to add the locale in the cacheKey when use the single-tree multilingual approach.

**Comment by kartobi** [comment]
Did you add cache property for locale (see http://documentation.opencms.org/opencms-documentation/caching-in-opencms/the-flex-cache/#2)

**Comment by tutotio** [comment]
I think the link cache is different from flex cache. The flex cache is for the jsp, but the link cache is used into the same jsp. If you run this code in a jsp:
```
<%@page buffer="none" session="false" trimDirectiveWhitespaces="true"%>
<%@ taglib prefix="cms" uri="http://www.opencms.org/taglib/cms"%>
<ul>
	<li>
		<a href="<cms:link locale='gl'>/</cms:link>">gl</a> 
	</li>
	<li>
		<a href="<cms:link locale='es'>/</cms:link>">es</a> 
	</li>
	<li>
		<a href="<cms:link locale='en'>/</cms:link>">en</a> 
	</li>
</ul>
```

the result is the same for the all locales (the first you run), gl in this case.

This code works fine in the offline Project but not in the online Project due to the 'if' in the line 233 in CmsDefaultLinkSubstitutionHandler.

**Comment by tHerrmann** [comment]
This has been fixed with commit 7dbc05ee90bc0a5cbeb7aaae40a54a51cf3c8621.

**Comment by tHerrmann** [comment]
The fix to this issue has been released with version [10.5.2](https://github.com/alkacon/opencms-core/releases/tag/build_10_5_2).
